src/components/TheHeader.vue

```diff
@@ -0,0 +1,97 @@
+<script lang="ts" setup>
+import MoonIcon from "@/assets/icons/MoonIcon.vue";
+import { EThemeMode } from "@/types/countries";
+import type { ComputedRef, Ref } from "vue";
+import { computed, ref } from "vue";
+
+const darkTheme = "dark-theme";
+const themeMode: Ref<EThemeMode> = ref(EThemeMode.LIGHT_MODE);
+const themeModeTitle: ComputedRef<string> = computed(() =>
+  themeMode.value === EThemeMode.LIGHT_MODE ? "Light Mode" : "Dark Mode"
+);
+
+const checkIfDarkTheme = () => localStorage.getItem(darkTheme);
+
+const toggleLocalStorageItem = () => {
+  if (checkIfDarkTheme()) {
+    localStorage.removeItem(darkTheme);
+  } else {
+    localStorage.setItem(darkTheme, "true");
+  }
+};
+
+const changeThemeMode = () => {
+  themeMode.value =
+    themeMode.value === EThemeMode.LIGHT_MODE
+      ? EThemeMode.DARK_MODE
+      : EThemeMode.LIGHT_MODE;
+
+  document.querySelector(":root")?.classList.toggle(darkTheme);
+  toggleLocalStorageItem();
+};
+
+if (checkIfDarkTheme()) {
+  themeMode.value = EThemeMode.DARK_MODE;
+  document.querySelector(":root")?.classList.toggle(darkTheme);
+}
```

Comment:
Rzeczy do togglowania theme'a wyniósłbym do hooka albo stworzył oddzielny komponent dla togglowania themem i tam zamknął całą logikę. Uprości to kod w samym headerze oraz wydzielisz logikę zmiany motywu, która nie dotyczy samego headera.